meshtastic/interfaces/ble/interface.py

```
@@ -2036,6 +1774,177 @@ def _sanitize_address(self, address: str | None) -> str | None:
+        self._client_manager.update_client_reference(client, previous_client)
+
+    @staticmethod
+    def _retry_policy_should_retry(policy: object, attempt: int) -> bool:
+        """Evaluate retry decision with public-first, underscore-fallback lookup."""
+        should_retry = getattr(policy, "should_retry", None)
+        if callable(should_retry):
+            return bool(should_retry(attempt))
+        legacy_should_retry = getattr(policy, "_should_retry", None)
+        if callable(legacy_should_retry):
+            return bool(legacy_should_retry(attempt))
+        raise AttributeError("Retry policy missing should_retry/_should_retry")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)